src/Symfony/Bundle/FrameworkBundle/Command/DebugAutowiringCommand.php

```diff
@@ -127,6 +127,9 @@ protected function execute(InputInterface $input, OutputInterface $output)
                     $serviceLine .= ' - <fg=magenta>deprecated</>';
                 }
             } elseif (!$all) {
+                if ($search) {
+                    $io->text('Non-aliased services are not shown unless the "--all" option is passed, you would better not type-hint on these classes.');
```

Comment:
this is going to be displayed once per not-displayed service right? It should be done just once for all services :)
The message could be e.g.
`X more concrete service(s) would be displayed when adding the --all option.`
Then, when `--all` is used, we could display something like:
`Please use interfaces in your type-hints instead of classes to benefit from the dependency inversion principle.`